iceberg/pom.xml

```diff
@@ -0,0 +1,325 @@
+<!--                        <path>-->
+<!--                            <groupId>com.google.errorprone</groupId>-->
+<!--                            <artifactId>error_prone_core</artifactId>-->
+<!--                            <version>${google.errorprone.version}</version>-->
+<!--                        </path>-->
+<!--                    </annotationProcessorPaths>-->
+<!--                </configuration>-->
+<!--            </plugin>-->
+
+            <plugin>
+                <groupId>com.diffplug.spotless</groupId>
+                <artifactId>spotless-maven-plugin</artifactId>
+                <version>${spotless.maven.plugin.version}</version>
+                <configuration>
+                    <java>
+                        <removeUnusedImports />
+                        <importOrder>
+                            <!-- normal imports, static imports -->
+                            <order>,\# </order>
+                        </importOrder>
+                        <trimTrailingWhitespace/>
+                        <endWithNewline/>
+                        <indent>
+                            <spaces>true</spaces>
+                            <spacesPerTab>2</spacesPerTab>
+                        </indent>
+                    </java>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>validate</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>check</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <version>${iceberg.checkstyle.plugin.version}</version>
+                <configuration>
+                    <propertyExpansion>config_loc=${basedir}/${path.to.iceberg.root}/checkstyle/</propertyExpansion>
+                    <configLocation>${basedir}/${path.to.iceberg.root}/checkstyle/checkstyle.xml</configLocation>
+                    <suppressionsLocation>${basedir}/${path.to.iceberg.root}/checkstyle/checkstyle-suppressions.xml</suppressionsLocation>
+                    <includeTestSourceDirectory>true</includeTestSourceDirectory>
+                    <encoding>UTF-8</encoding>
+                    <consoleOutput>true</consoleOutput>
+                    <failsOnError>true</failsOnError>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>validate</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>check</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+<!--    <profiles>-->
```

Comment:
Isn't it enough to keep this either here or just in the child modules?

Reply:
Might worth to check if we want to keep it.

We should spend some time on the errorprone plugin and decide what to do with it. I was not able to compile the code reliably in IntelliJ with the plugin enabled.

Reply:
As long as the upstream Iceberg community keeps the baseline plugins, I think we should have it too. But I agree it's painful with IntelliJ.

Perhaps we can hide it behind a profile flag and run it only on demand? I wonder if there's some profile/property the CI builds use which we could use to activate this only during CI runs. That would mean we'd get to run it locally only on demand (therefore not messing up the IDE) but still enforce the validation checks as a precommit requirement.

Reply:
If there's nothing better, we could perhaps use the `dist` profile and run the errorprone checks only then

Reply:
Used a profile to handle `errorProne`.

See http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-2174/1/pipeline/

Reply:
Perfect!